csharp/src/Google.Protobuf/JsonParser.cs

```diff
@@ -67,15 +67,18 @@ public sealed class JsonParser
 
         private static readonly JsonParser defaultInstance = new JsonParser(Settings.Default);
 
-        private static readonly Dictionary<string, Action<JsonParser, IMessage, JsonTokenizer>>
-            WellKnownTypeHandlers = new Dictionary<string, Action<JsonParser, IMessage, JsonTokenizer>>
+        // TODO: Consider introducing a class containing parse state of the parser, tokenizer and depth. That would simplify these handlers
```

Comment:
Yes, that sounds good if you don't mind.

I'd expect the resulting code to be simpler and less prone to forgetting to pass depth arg somewhere.